compiler/ilgen/MethodBuilder.cpp

```diff
@@ -349,15 +352,16 @@ MethodBuilder::lookupSymbol(const char *name)
       return (TR::SymbolReference *)_symbols->getData(symbolsID);
 
    TR::SymbolReference *symRef;
-   TR_HashId typesID;
-   _symbolTypes->locate(name, typesID);
+   SymbolTypeMap::iterator symTypesIterator =  _symbolTypes.find(name);
+
+   TR_ASSERT_FATAL(symTypesIterator != _symbolTypes.end(), "Symbol '%s' doesn't exist", name);
 
-   TR::DataType type = ((TR::IlType *)(_symbolTypes->getData(typesID)))->getPrimitiveType();
+   TR::DataType type = symTypesIterator->second->getPrimitiveType();
```

Comment:
I confess I'm not a big fan of these iterator->second references. Would it be too painful to put them into a variable of the same type, even if you just use auto? I just loathe templates when one tries to debug this code, so if I can avoid grubbing around in a std::pair, that feels worth it to me.